### PR TITLE
UCP/TEST: Call start_(un)pack of generic datatypes immediately.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1063/ for details.

**Comment by shamisp** [comment]
autobot build

File: src/ucp/api/ucp.h [hunk 1]
```diff
@@ -325,6 +325,8 @@ typedef struct ucp_config {
     size_t                                 bcopy_thresh;
     /** Threshold for switching UCP to rendezvous protocol */
     size_t                                 rndv_thresh;
+    /** How much packet data to dump in debug mode */
```
Comment (shamisp) [inline src/ucp/api/ucp.h]: /** The size (in bytes) of the log buffer that is used for a logging of the network packet information (headers, payload, etc.). The information that does not fit into the log buffer will be truncated.*/

File: src/ucp/core/ucp_context.c [hunk 2]
```diff
@@ -57,6 +57,10 @@ static ucs_config_field_t ucp_config_table[] = {
    "Threshold for switching from eager to rendezvous protocol",
    ucs_offsetof(ucp_config_t, rndv_thresh), UCS_CONFIG_TYPE_MEMUNITS},
 
+  {"LOG_DATA", "0",
+   "How much of packet payload to dump in debug mode (0 - disable).",
```
Comment (shamisp) [inline src/ucp/core/ucp_context.c]: "The size (in bytes) of the log buffer that is used for a logging of the network packet information (headers, payload, etc.). The information that does not fit into the log buffer will be truncated."

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/490/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
Reviewed. Functionality wise I don't have concerns about this patch. In terms of documentation, please see my comments.

File: src/ucp/api/ucp.h (hunk 1 above)
Comment (yosefe) [inline src/ucp/api/ucp.h]: this is not correct documentation. It's how much packed data to dump,not the log buffer size.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1065/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/494/ for details (Mellanox internal link).

**Comment by miked-mellanox** [comment]
:+1:

File: src/ucp/api/ucp.h [hunk 3]
```diff
@@ -325,6 +325,8 @@ typedef struct ucp_config {
     size_t                                 bcopy_thresh;
     /** Threshold for switching UCP to rendezvous protocol */
     size_t                                 rndv_thresh;
+    /** How much of the packet data to dump in debug mode */
```
Comment (shamisp) [inline src/ucp/api/ucp.h]: The size of the packed data that is dumped to the log system in a debug mode.

File: src/ucp/core/ucp_context.c [hunk 4]
```diff
@@ -57,6 +57,10 @@ static ucs_config_field_t ucp_config_table[] = {
    "Threshold for switching from eager to rendezvous protocol",
    ucs_offsetof(ucp_config_t, rndv_thresh), UCS_CONFIG_TYPE_MEMUNITS},
 
+  {"LOG_DATA", "0",
+   "How much of packet payload to dump in debug mode (0 - nothing).",
+   ucs_offsetof(ucp_config_t, log_data_size), UCS_CONFIG_TYPE_MEMUNITS},
```
Comment (shamisp) [inline src/ucp/core/ucp_context.c]: The size of the packed data that is dumped to the log system in debug mode.

**Comment by shamisp** [comment]
:+1:

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1066/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/495/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1067/ for details.

**Pull request merged**